docs: Mention compression header behaviour in `reverse_proxy` (#193)
caddyserver/website: src/docs/markdown/caddyfile/directives/reverse_proxy.md

- It adds or augments the [X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) header field.
- It sets the [X-Forwarded-Proto](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) header field.

> Additionally, Caddy will also set `Accept-Encoding: gzip` if that header is missing in the request from the client. This behavior can be disabled by setting `compression off` in the `http` transport.

Reviewer: ```suggestion
Additionally, when using the [`http` transport](#the-http-transport), the `Accept-Encoding: gzip` header will be set, if it is missing in the request from the client. This behavior can be disabled with [`compression off`](#compression) on the transport.
```
